src/main/java/com/breskul/hw/controllers/NasaPictureController.java

```diff
@@ -0,0 +1,57 @@
+package com.breskul.hw.controllers;
+
+import com.breskul.hw.endpoints.dto.LargestNasaPictureRequest;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.amqp.rabbit.core.RabbitTemplate;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+@Component
+public class NasaPictureController {
```

Comment:
Calling it as `Controller` can be a bit confusing in my opinion, I recommend name it as `Service` as it is responsible for business logic and integrations.